### PR TITLE
Chore(experiment): Adds AWS ECS experiment docs

### PR DESCRIPTION
**Pull request by uditgaurav**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      <tr> 
+        <td> DOCUMENT_TYPE </td>
+        <td> Provide the document type of added ssm docs (if not using the default docs)</td>
+        <td> Defaults to <code>Command</code> </td>
+      </tr>
+      <tr> 
+        <td> DOCUMENT_PATH </td>
+        <td> Provide the document path if added using configmaps</td>
+        <td> Defaults to Litmus-AWS-SSM-Docs-For-CPU-Hog.yml </td>
+      </tr>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md:129]: as suggested by @avaakash "We can remove these as the default document will be used for these experiments, if the user wants to use a custom document they can use the SSM Command experiment ( as discussed recently)"

File: docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      <tr> 
+        <td> INSTALL_DEPENDENCIES </td>
+        <td> Select to install dependencies used to run the cpu chaos. It can be either True or False</td>
+        <td> Defaults to True </td>
+      </tr>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md]: as suggested by @avaakash "We can remove these as the default document will be used for these experiments, if the user wants to use a custom document they can use the SSM Command experiment ( as discussed recently)"

Comment (ispeakc0de) [reply docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md]: check other places too

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-io-stress.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+        # path need to be stressed/filled
+        - name: VOLUME_MOUNT_PATH
+          value: '/some-dir-in-container'
+        - name: TOTAL_CHAOS_DURATION
+          VALUE: '60'
+```
+
+### Workers For Stress
+
+The worker's count for the stress can be tuned with NUMBER_OF_WORKERS ENV.
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-io-stress.md]:
```suggestion
The worker's count for the stress can be tuned with `NUMBER_OF_WORKERS` ENV.
```

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-network-lantecy.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+        <td> Provide the document path if added using configmaps</td>
+        <td> Defaults to Litmus-AWS-SSM-Docs-For-memory-hog.yml </td>
+      </tr>
+      <tr> 
+        <td> INSTALL_DEPENDENCIES </td>
+        <td> Select to install dependencies used to run the Memory chaos. It can be either True or False</td>
+        <td> Defaults to True </td>
+      </tr>
+      <tr> 
+        <td>  NETWORK_LATENCY </td>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-network-lantecy.md]: NETWORK_INTERFACE, DESTINATION_IPS, and  DESTINATION_HOSTS  envs are missing - same in network-loss

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-network-lantecy.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
+---
+id: ecs-container-network-latency
+title: ECS Container Network Latency
+---
+
+## Introduction
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-network-lantecy.md]: typo in the file name lantecy --> latency

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-network-lantecy.md (hunk 4 above)
Comment (ispeakc0de) [reply docs/chaos-engineering/chaos-faults/aws/ecs-container-network-lantecy.md]: add these entries into table of .png image as well

File: docs/chaos-engineering/chaos-faults/chaos-faults.md [hunk 6]
```diff
@@ -260,6 +260,31 @@ Following Platform Chaos faults are available:
     <td>Injects dns faults on the target ec2 instance(s)</td>
     <td><a href="/docs/chaos-engineering/chaos-faults/aws/ec2-dns-chaos">ec2-dns-chaos</a></td>
   </tr>
+  <tr>
+    <td>ECS Container CPU Hog</td>
+    <td>Injects container cpu hog chaos on task container</td>
+    <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog">ec2-dns-chaos</a></td>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/chaos-faults.md]:
```suggestion
    <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog">ecs-container-cpu-hog</a></td>
```

File: docs/chaos-engineering/chaos-faults/chaos-faults.md [hunk 7]
```diff
@@ -260,6 +260,31 @@ Following Platform Chaos faults are available:
...
+  </tr>
+  <tr>
+    <td>ECS Container IO Stress</td>
+    <td>Injects container IO stress chaos on task container</td>
+    <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-container-io-stress">ec2-dns-chaos</a></td>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/chaos-faults.md:271]:
```suggestion
    <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-container-io-stress">ecs-container-io-stress</a></td>
```

File: docs/chaos-engineering/chaos-faults/chaos-faults.md [hunk 8]
```diff
@@ -260,6 +260,31 @@ Following Platform Chaos faults are available:
...
+  </tr>
+  <tr>
+    <td>ECS Container Memory Hog</td>
+    <td>Injects container memory hog chaos on task container</td>
+    <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-container-memory-hog">ec2-dns-chaos</a></td>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/chaos-faults.md:276]:
```suggestion
    <td><a href="/docs/chaos-engineering/chaos-faults/aws/ecs-container-memory-hog">ecs-container-memory-hog</a></td>
```

File: docs/chaos-engineering/chaos-faults/chaos-faults.md [hunk 9]
```diff
@@ -260,6 +260,31 @@ Following Platform Chaos faults are available:
...
+  </tr>
+  <tr>
+    <td>EC2 Network Latency</td>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/chaos-faults.md]:
```suggestion
    <td>ECS Container Network Latency</td>
```

Comment (ispeakc0de) [reply docs/chaos-engineering/chaos-faults/chaos-faults.md]: check other names and filepaths

File: docs/chaos-engineering/chaos-faults/chaos-faults.md (hunk 6 above)
Comment (ispeakc0de) [reply docs/chaos-engineering/chaos-faults/chaos-faults.md]: check other places too

File: docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+## Experiment tunables
+
+<details>
+    <summary>Check the Experiment Tunables</summary>
+    <h2>Mandatory Fields</h2>
+    <table>
+        <tr>
+        <th> Variables </th>
+        <th> Description </th>
+        <th> Notes </th>
```
Comment (avaakash) [inline docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md]:
```suggestion
            <th> Variables </th>
            <th> Description </th>
            <th> Notes </th>
```
Can we add indentations in HTML code

File: docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+        </tr>
+        <tr> 
+        <td> CLUSTER_NAME </td>
+        <td> Name of the target ECS cluster</td>
+        <td>  </td>
```
Comment (avaakash) [inline docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md]: We can add an example in the notes section (or default value) to maintain consistency in docs.

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+:::tip Fault execution flow chart
+![ECS Container CPU Hog](./static/images/ecs-stress-chaos.png)
+:::
+
+## Uses
+
+<details>
+<summary>View the uses of the experiment</summary>
+<div>
+Disk Pressure or CPU hogs is another very common and frequent scenario we find with conrainers/applications that can result in the eviction of the application (task container) and impact its delivery. Such scenarios that can still occur despite whatever availability aids docker provides. These problems are generally referred to as "Noisy Neighbour" problems.
```
Comment (avaakash) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md]:
```suggestion
CPU hogs are another very common and frequent scenario we find with containers/applications that can result in the eviction of the application (task container) and impact its delivery. Such scenarios can still occur despite whatever availability aids docker provides. These problems are generally referred to as "Noisy Neighbour" problems.
```

File: docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+:::tip Fault execution flow chart
+![ECS Agent Stop](./static/images/ecs-agent-stop.png)
+:::
+
+## Uses
+
+<details>
+<summary>View the uses of the experiment</summary>
+<div>
+Agent chaos stop is another very common and frequent scenario we find with ECS clusters that can result in breaking of agent that manages task container on ECS cluster and impact its delivery. Such scenarios that can still occur despite whatever availability aids docker provides.
```
Comment (avaakash) [inline docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md]:
```suggestion
Agent chaos stop is another very common and frequent scenario we find with ECS clusters that can break an agent that manages the task container on the ECS cluster and impacts their delivery. Such scenarios can still occur despite whatever availability aids docker provides.
```

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md [hunk 14]
```diff
@@ -0,0 +1,227 @@
+---
+id: ecs-container-cpu-hog
+title: ECS Container CPU Hog
+---
+
+## Introduction
+
+- ECS Container CPU hog contains chaos to disrupt the state of infra resources. The experiment can induce a stress chaos on AWS ECS container using Amazon SSM Run Command, this is carried out by using SSM Docs which is in-built in the experiment for the give chaos experiment.
```
Comment (avaakash) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md]:
```suggestion
- ECS Container CPU hog contains chaos to disrupt the state of infra resources. The experiment can induce stress chaos on the AWS ECS container using Amazon SSM Run Command, this is carried out by using SSM Docs which is in-built into the experiment for the give chaos experiment.
```

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-memory-hog.md [hunk 15]
```diff
@@ -0,0 +1,252 @@
...
+:::tip Fault execution flow chart
+![ECS Container Memory Hog](./static/images/ecs-stress-chaos.png)
+:::
+
+## Uses
+
+<details>
+<summary>View the uses of the experiment</summary>
+<div>
+Memory usage within containers is subject to various constraints. If the limits are specified in their spec, exceeding them can cause termination of the container (due to OOMKill of the primary process, often pid 1) - the restart of the container by docker, subject to the policy specified. For containers with no limits placed, the memory usage is uninhibited until such time as the VM level OOM Behaviour takes over. In this case, containers on the Instance can be killed based on their oom_score. This eval is extended to all task container running on the instance - thereby causing a bigger blast radius.
```
Comment (avaakash) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-memory-hog.md]:
```suggestion
Memory usage within containers is subject to various constraints. If the limits are specified in their spec, exceeding them can cause termination of the container (due to OOMKill of the primary process, often pid 1) - the restart of the container by docker, subject to the policy specified. For containers with no limits placed, the memory usage is uninhibited until such time as the VM level OOM Behaviour takes over. In this case, containers on the Instance can be killed based on their oom_score. This eval is extended to all task containers running on the instance - thereby causing a bigger blast radius.
```

**Review by avaakash — CHANGES_REQUESTED** [review]

**Comment by uditgaurav** [comment]
@ispeakc0de / @avaakash Have addressed the comments

File: docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+      </tr>
+      <tr>
+        <td> SEQUENCE </td>
+        <td> It defines sequence of chaos execution for multiple instance</td>
+        <td> Default value: parallel. Supported: serial, parallel </td>
+      </tr>
+      <tr>
+        <td> RAMP_TIME </td>
+        <td> Period to wait before and after injection of chaos in sec </td>
+        <td> </td>
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/aws/ecs-agent-stop.md]:
```suggestion
        <td> Eg. 30 </td>
```

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+    <table>
+        <tr>
+        <th> Variables </th>
+        <th> Description </th>
+        <th> Notes </th>
+        </tr>
+        <tr> 
+        <td> CLUSTER_NAME </td>
+        <td> Name of the target ECS cluster</td>
+        <td>  </td>
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md]:
```suggestion
        <td> Eg. cluster-1 </td>
```

File: docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+        </tr>
+        <tr>
+        <td> REGION </td>
+        <td> The region name of the target ECS cluster</td>
+        <td> </td>
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/aws/ecs-container-cpu-hog.md]:
```suggestion
        <td> Eg. us-east-1 </td>
```

**Review by neelanjan00 — APPROVED** [review]

**Review by avaakash — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**